File: lib/src/user_profile.dart
part of flutter_line_sdk;

class UserProfile {
  UserProfile._(this._data);

  final Map<String, dynamic> _data;

  String get displayName => _data['displayName'];
  String get userId => _data['userId'];
  String get statusMessage => _data['statusMessage'];
  String get pictureUrl => _data['pictureUrl'];
  String get pictureUrlLarge => _data['pictureUrl'] + '/large';
  String get pictureUrlSmall => _data['pictureUrl'] + '/small';
}

File: android/src/main/kotlin/com/linecorp/flutter_line_sdk/model/UserProfile.kt
package com.linecorp.flutter_line_sdk.model

import androidx.annotation.Keep
import com.linecorp.linesdk.LineProfile

@Keep
data class UserProfile(
    val userId: String,
    val displayName: String,
    val pictureUrl: String,
    val statusMessage: String
) {
    companion object {
        fun convertLineProfile(profile: LineProfile): UserProfile =
            UserProfile(
                profile.userId,
                profile.displayName,
                profile.pictureUrl?.toString() ?: "",
                profile.statusMessage ?: ""
            )
    }
}

